### PR TITLE
[Builds on 1611]  Refactor compact blocks out of CNode

### PR DESCRIPTION
**Pull request by ptschip**
Also consolidates some additional common functions between thinblocks and compact blocks.  Will do the same when the refactor for graphene happens.

Just need to add some tests.

### CODE REVIEW

**Comment by ptschip** [comment]
taking out of WIP

**Comment by ptschip** [comment]
rebased

**Comment by sickpig** [comment]
@ptschip 

compile error: 

```
  CXX      blockrelay/libbitcoin_server_a-thinblock.o
  CXX      blockstorage/libbitcoin_server_a-blockleveldb.o
../../src/blockrelay/thinblock.cpp:513:22: error: no member named 'ClearThinBlockData' in 'CThinBlockData'
            thindata.ClearThinBlockData(pfrom, pblock);
            ~~~~~~~~ ^
1 error generated.
```

**Comment by ptschip** [comment]
rebased

File: src/blockrelay/blockrelay_common.cpp [hunk 1]
```diff
@@ -280,3 +286,83 @@ void ThinTypeRelay::ClearBlockToReconstruct(CNode *pfrom)
...
+            nLargestBytes = nBytes;
+        }
+    }
+
+    if (pLargestNode != nullptr)
+    {
+        thinrelay.ClearAllBlockData(pLargestNode, pLargestBlock);
+        pLargestNode->fDisconnect = true;
+
+        // If the our node is currently using up the most thinblock bytes then return true so that we
```
Comment (sickpig) [inline src/blockrelay/blockrelay_common.cpp:317]: If our node

File: src/blockrelay/blockrelay_common.cpp [hunk 2]
```diff
@@ -280,3 +286,83 @@ void ThinTypeRelay::ClearBlockToReconstruct(CNode *pfrom)
     LOCK(cs_reconstruct);
     mapBlocksReconstruct.erase(pfrom->GetId());
 }
+
+bool ThinTypeRelay::ClearLargestBlockAndDisconnect(CNode *pfrom)
```
Comment (sickpig) [inline src/blockrelay/blockrelay_common.cpp:290]: I wonder if we could add a comment to explain a bit what this function is goign to do

**Review by sickpig — APPROVED** [review]
approved. just left 2 comments about 2 very minor nits

**Pull request merged**